src/jobs/request.jl

```diff
@@ -0,0 +1,92 @@
+"""
+    function request(
+        job::Job,
+        method::AbstractString,
+        uripath::AbstractString,
+        [body];
+        [auth::Authentication],
+        [extra_headers],
+        kwargs...
+    ) -> HTTP.Response
+
+Performs a HTTP against the HTTP server exposed by the job with the authentication
+token of the authenticated user. The function is a thin wrapper around the `HTTP.request`
+function, constructing the correct URL and setting the authentication headers.
+
+Arguments:
+
+* `job::Job`: JuliaHub job (from [`JuliaHub.job`](@ref))
+
+* `method::AbstractString`: HTTP method (gets directly passed to HTTP.jl)
+
+* `uripath::AbstractString`: the path and query portion of the URL, which gets
+  appended to the scheme and hostname port of the URL. Must start with a `/`.
+
+* `body`: gets passed as the `body` argument to HTTP.jl
+
+Keyword arguments:
+
+$(_DOCS_authentication_kwarg)
+
+* `extra_headers`: an iterable of extra HTTP headers, that gets concatenated
+  with the list of necessary authentication headers and passed on to `HTTP.request`.
+
+  * Additional keyword arguments must be valid HTTP.jl keyword arguments and will
+  get directly passed to the `HTTP.request` function.
+
+!!! note
+
+    See the [manual section on exposing ports](@ref jobs-batch-expose-port) and
+    the `expose` argument to [`submit_job`](@ref).
+"""
+function request(
+    job::Job,
+    method::AbstractString,
+    uripath::AbstractString,
+    body::Any = UInt8[];
+    auth::Authentication=__auth__(),
```

Comment:
Should we allow for `::Nothing` or similar here as well? To allow for "anonymous" access, basically.

Reply:
Also, the function is a very thin wrapper around `HTTP.request`. So if you do want to make unauthenticated requests, you can just get the hostname from `job.hostname` and call `HTTP.request` yourself.